### PR TITLE
Ensure SHORT local props are handled properly.

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 
Ensure SHORT configuration items set in the CLI are correctly handled by the KSQL server.

Fixes: #2099

See the above issues for details.  Brief overview of the problem is that the current KSQL Server can't handle SHORT properties, e.g. `ksql.sink.replicas`, being overridden in a request to the server where the value is supplied as a numeric, rather than a string e.g. the following will result in an error:

```
{
   ksql: "Blah",
   streamsProperties: {
       ksql.sink.replicas: 2
   }
}
```

Because the number is deserialised by the Jackson JSON deserializer to `Integer(2)`, but the type of the property is a `SHORT`. `ConfigDef` can't handle coercing an integer to a short and so complains the property can't be set to the supplied version.

This worked in previous versions for `ksql.sink.replicas` specifically, (though not for other SHORT configs), because the CLI passed `KsqlConfig` values as a `String`, i.e.

```
{
   ksql: "Blah",
   streamsProperties: {
       ksql.sink.replicas: "2"
   }
}
```

Which the `ConfigDef` could coerce to a short.

One potential fix would be to have the CLI continue to send the value as a String. However, this would leave the underlying issue in the server.

Hence, this PR looks to enhance the server to support correctly parsing / coercing of a numeric value to the short property. It does this by simply converting it to a string before passing it to `ConfigDef`.

### Testing done 
Added tests + manual testing.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java [hunk 1]
```diff
@@ -55,9 +53,21 @@ public Object parse(final String property, final Object value) {
...
     validator.validate(configItem.getPropertyName(), parsedValue);
     return parsedValue;
   }
+
+  private Object parse(final Object value, final ConfigItem configItem) {
+    final boolean isShortProperty = configItem.getPropertyType()
+        .map(type -> type == Type.SHORT)
+        .orElse(false);
+
+    if (isShortProperty) {
```
Comment (dguy) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java]: why not just always do `configItem.parseValue(String.valueOf(value))`?

Comment (apurvam) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java]: +1, if we know that strings can be safely coerced to the underlying type, why not only pass strings? At the very least, we should document the reason why short is special cased and not other types. Is short the only one which can't be coerced?

Comment (big-andy-coates) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java]: Two reasons:
- Why pay the price of Object -> String -> Object.
- Why introduce potentially new issues with the way ConfigDef parses Strings?

@apurvam - the documentation you request is already there. check the comment on the line below.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import static org.hamcrest.Matchers.is;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.junit.Test;
+
+public class KsqlRequestTest {
+
+  @Test
+  public void shouldCoerceShortPropertyValues() {
```
Comment (hjafarpour) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlRequestTest.java]: Shall we have some tests for other data types too?

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java (hunk 1 above)
Comment (dguy) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java]: @big-andy-coates the price? it isn't exactly on the critical path - so i don't see it as an issue at all. We already know that ConfigDef parses Strings - that is what it generally does...

Comment (big-andy-coates) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java]: True.  OK, convinced.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlRequestTest.java (hunk 2 above)
Comment (big-andy-coates) [reply ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlRequestTest.java]: More tests added.

**Pull request merged**